go/src/koding/api/cache.go

```diff
@@ -117,49 +104,52 @@ func (s *SessionCache) Auth(opts *AuthOptions) (*Session, error) {
 	// fail are filesystem errors - there is no recovery from
 	// that, and falling back to memory makes sense only for
 	// Klient.
-	s.cacheMu.Lock()
 	if opts.Refresh {
-		_ = s.delete(session)
+		_ = s.Storage.Delete(session)
 	}
 	if err == nil {
-		_ = s.set(session)
+		_ = s.Storage.Set(session)
 	}
-	s.cacheMu.Unlock()
 
 	return session, err
 }
 
-func (s *SessionCache) get(session *Session) error {
-	if s.Storage != nil {
-		return s.Storage.Get(session)
-	}
+type defaultStorage struct {
+	cache cache.Cache
+}
 
-	sess, ok := s.cache[session.Key()]
-	if !ok {
-		return ErrSessionNotFound
-	}
+var _ Storage = (*defaultStorage)(nil)
 
-	*session = *sess
+func newDefaultStorage() *defaultStorage {
+	n := DefaultCacheSize * 1024 * 1024 / int(unsafe.Sizeof(Session{}))
```

Comment:
this "n" is more than 11M :) ((256*1024*1024)/24)

what about decreasing it to 2K? (based on usage)